tests/framework/e2e/downgrade.go

```diff
@@ -162,15 +163,34 @@ func DowngradeUpgradeMembersByID(t *testing.T, lg *zap.Logger, clus *EtcdProcess
 		}
 	}
 
-	clusterVersion := targetVersion.String()
-	if !isDowngrade && len(membersToChange) != len(clus.Procs) {
-		clusterVersion = currentVersion.String()
-	}
+	t.Log("Waiting health interval to make sure the leader propagates version to new processes")
+	time.Sleep(etcdserver.HealthInterval)
+
 	lg.Info("Validating versions")
+	clusterVersion := targetVersion
+	if !isDowngrade {
+		// if there is any member using old version binary, the clusterVersion should be old version.
+		for _, member := range clus.Procs {
+			mVersion, err := GetVersionFromBinary(member.Config().ExecPath)
+			if err != nil {
+				return fmt.Errorf("failed to get version from member %s binary: %w", member.Config().Name, err)
+			}
+
+			if mVersion.LessThan(*clusterVersion) {
+				clusterVersion = mVersion
+				break
+			}
+		}
+
+		if downgradeEnabled {
+			clusterVersion = currentVersion
+		}
+	}
```

Comment:
```suggestion
	if !isDowngrade {
		if downgradeEnabled {
			// If the downgrade isn't cancelled yet, then the cluster
			// version will always stay at the lower version, no matter
			// what's the binary version of each member.
			clusterVersion = currentVersion
		} else {
			// If the downgrade has already been cancelled, then the
			// cluster version is the minimal server version.
			minVer, err := clus.MinServerVersion()
			if err != nil {
				return fmt.Errorf("failed to get min server version: %w", err)
			}
			clusterVersion = minVer
		}
	}	
```

Also add a method below,

```
func (epc *EtcdProcessCluster) MinServerVersion() (*semver.Version, error) {
	var minVersion *semver.Version
	for _, member := range epc.Procs {
		ver, err := GetVersionFromBinary(member.Config().ExecPath)
		if err != nil {
			return nil, fmt.Errorf("failed to get version from member %s binary: %w", member.Config().Name, err)
		}

		if minVersion == nil || ver.LessThan(*minVersion) {
			minVersion = ver
		}
	}
	return minVersion, nil
}
```